qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/plot_presenter.py

```diff
@@ -112,65 +110,17 @@ def do_serial_fit(self):
-            table_dict.pop("SpecIndex", None)
-            return {col_name.split("_")[-1]: value for col_name, value in table_dict.items()}
-        return None
-
     def run_find_peaks_convolve(self):
         self.is_waiting_convolve_peaks = False
         self.find_peaks_convolve_started_notifier.notify_subscribers()
         try:
             input_ws_name = self.view.fit_browser.workspaceName()
-            if not ADS.doesExist(input_ws_name):
-                logger.error(f"{input_ws_name} does not exist!")
-                self.find_peaks_convolve_done_notifier.notify_subscribers(False)
-                return
-            fit_prop_ws = ADS.retrieve(input_ws_name)
-            groupWs = FindPeaksConvolve(
-                InputWorkspace=fit_prop_ws,
-                OutputWorkspace="FindPeaksConvolve_" + input_ws_name,
-                IOverSigmaThreshold=4,
-                EstimatedPeakExtent=200,
-                StoreInADS=True,
-            )
-            peak_x_values = dict()
-            peak_y_values = dict()
-            for tab_ws in groupWs:
-                if tab_ws.getName() == "PeakCentre":
-                    peak_x_values = self._re_organize_keys_find_peaks_convolve(tab_ws)
-                elif tab_ws.getName() == "PeakYPosition":
-                    peak_y_values = self._re_organize_keys_find_peaks_convolve(tab_ws)
-
-            if peak_x_values is None or peak_y_values is None:
-                logger.error("Failed extracting columns from FindPeakConvolve output")
-                self.find_peaks_convolve_done_notifier.notify_subscribers(False)
-                return
-
-            if set(peak_x_values.keys()) == set(peak_y_values.keys()):
-                logger.notice(f"Adding {len(peak_x_values)} peaks found via FindPeaksConvolve")
-                func_wrapper = None
-                for col, x_value in peak_x_values.items():
-                    y_value = peak_y_values[col]
-                    peak_func = FunctionFactory.Instance().createPeakFunction(self.view.fit_browser.defaultPeakType())
-                    peak_func.setCentre(x_value)
-                    peak_func.setHeight(y_value)
-                    peak_func.setMatrixWorkspace(fit_prop_ws, 0, 0, 0)
-                    f_wrapper = FunctionWrapper(peak_func)
-                    if func_wrapper is None:
-                        func_wrapper = f_wrapper
-                    else:
-                        func_wrapper += f_wrapper
-                if func_wrapper is not None:
-                    self.is_waiting_convolve_peaks = True
-                    self.view.fit_browser.loadFunction(str(func_wrapper))
-                else:
-                    logger.error("No peak functions were created from the results of FindPeaksConvolve")
-                    self.find_peaks_convolve_done_notifier.notify_subscribers(False)
+            func_wrapper_str = self.model.run_find_peaks_convolve(input_ws_name, self.view.fit_browser.defaultPeakType())
```

Comment:
Would it be possible to only add peaks that are in the fit range defined by the green slider?
Atm peaks are added outside but the cost function will not depend on the paramters of these peaks.
![image](https://github.com/mantidproject/mantid/assets/55979119/65519343-9f8e-4e74-b3d1-8642f70fb6ab)

 This would probably involve passing a min and max x here?

Reply:
I have updated the PR with this change. Thank you!